### PR TITLE
Methods to stitch a position grid

### PR DESCRIPTION
**Pull request by ieivanov**
She is ready!

This PR adds methods to stitch a grid of positions. Stitch parameters are computed with `>>> mantis estimate-stitch`. Images are then stitched with `>>> mantis stitch`. Image registration parameters are computed on pairs of images using phase cross-correlation; the pipeline can be extended to include more complex transformations. The image mosaic is compiled using "average" blending; other blending modes may be implemented later.

See examples in `/hpc/projects/intracellular_dashboard/ops/2024_03_05_registration_test/live/3-stitch/` and `/hpc/projects/intracellular_dashboard/ops/2024_04_11_Manual_HELA/1-stitch/v3/`

Replaces #135 
See https://github.com/czbiohub-sf/shrimPy/pull/135#issuecomment-2038005161

TODO:
 - [x] Convert `compute_image_translation` to CLI and save in config file
 - [x] Test with non-square image and non-square grids

### CODE REVIEW

File: mantis/cli/stitch.py
Comment (talonchandler) [inline mantis/cli/stitch.py:36]:
```suggestion
```
I'll suggest deleting this and the `if not slurm` block, since this option isn't really optional.

File: mantis/cli/stitch.py
Comment (talonchandler) [inline mantis/cli/stitch.py:50]:
```suggestion
```

File: mantis/cli/stitch.py
Comment (talonchandler) [inline mantis/cli/stitch.py:33]: Hmmm I didn't have default write access here. Do you know if there's an everyone-accessible `/tmp/`? Or could the default be `./`?

Doesn't need to be solved now...for now I'll suggest adding `--temp-path` to the docstring.

File: mantis/cli/stitch.py
Comment (talonchandler) [inline mantis/cli/stitch.py:47]:
```suggestion
    >>> mantis stitch -i ./input.zarr/*/*/* -c ./stitch_params.yml -o ./output.zarr --temp-dir ./
```

**Review by talonchandler — APPROVED** [review]
Big job! Looks great. I like the slurm job dependencies. 

I'm in the middle of recreating your `2024_03_05_registration_test/live/3-stitch/` (~20% through the final step filling the last store) and everything's going smoothly. 

Here are the small snags I've run into:
- `--slurm` parameter tripped me up on the `stitch` call. Seems required, so I suggest dropping it for now?
- the default `--temp-dir` wasn't writeable for me. Default to `./`?
- When I open the stitched zarr in napari I get `GL_MAX_TEXTURE_SIZE 16384 in at least one axis and will be downsampled.`, and the viewable resolution is noticeably worse than the originals. Do we need pyramids? Another trick? I don't think this needs to block this merge, though.

File: mantis/cli/stitch.py
Comment (ieivanov) [reply mantis/cli/stitch.py:33]: That's quite strange - I thought all of CompMicro should have access to `/hpc/scratch/group.comp.micro/`. Maybe check with IT? It's possible that you were trying to overwrite one of my temp files?

It's important that all jobs have access to this directory, that's why `/tmp` on our HPC is not a good choice since that space is local and different per node. As you suggest, `./` may be a better more general default, with a suggestion to use `/hpc/scratch/group.comp.micro/` in the docstring

**Comment by ieivanov** [comment]
Thanks for your review @talonchandler! I implemented your suggestions for `--temp-path` and `--slurm` options. Let's merge this and we'll fix other issues through smaller PRs.

>When I open the stitched zarr in napari I get GL_MAX_TEXTURE_SIZE 16384 in at least one axis and will be downsampled., and the viewable resolution is noticeably worse than the originals. Do we need pyramids? Another trick? I don't think this needs to block this merge, though.

That's true - so far this only happens for our largest (OPS) datasets and it hasn't bothered me much. It'll be nice to have a proper fix, which I think is implementing pyramids. This is not a blocker for now - analysis still happens on a FOV basis and detailed inspection can be done on smaller ROIs by loading a selection of the data through python.

File: mantis/cli/stitch.py
Comment (talonchandler) [reply mantis/cli/stitch.py]: Aha! My mistake. 

I just looked closer and found that you own `/TEMP_A3_600k_38x38_1.zarr`, so when I copied/reran your scripts I got permissions errors from that directory, not `/hpc/scratch/group.com.micro/`.

**Comment by talonchandler** [comment]
LGTM! Thanks @ieivanov.

**Pull request merged**
